csi/util.go

```diff
@@ -239,6 +239,24 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	return recurringJobs, nil
 }
 
+func syncMountPointDirectory(targetPath string) error {
+	d, err := os.Open(targetPath)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	if _, err := d.Readdirnames(-1); err != nil {
+		return err
+	}
+
+	if err := d.Sync(); err != nil {
+		return err
+	}
```

Comment:
Why need the `Sync` operation?

Reply:
It needs some I/O to trigger the mount point to be invalid.
Almost `I/O error` errors return from this `Sync()` method.

Reply:
Quite confusing about the operations here. Also, need some clarification.

Reply:
> It needs some I/O to trigger the mount point to be invalid. Almost `I/O error` errors return from this `Sync()` method.

Can't `Readdirnames` return IO error if the mountpoint is broken?

Reply:
It needs some I/Os to make the mount point invalid when volume detached unexpectedly.
1. Readdirnames(-1) would make mount point invalid. (it would return `I/O error` if there are other I/Os before this metohd.)
2. Sync() would return `I/O error`

Reply:
I see. You probably can refer to `ensureMountpoint()`. The function checks the availability of a mountpoint by `os.ReadDir`. Using the same method can make the codes more consistent.

Reply:
I see. You probably can refer to ensureMountpoint(). The function checks the availability of a mountpoint by os.ReadDir. Using the same method can make the codes more consistent.

Reply:
Another question:

> Readdirnames(-1) would make mount point invalid.

What error will be returned if mount point is invalid or become invalid when executing `Readdirnames`?

Reply:
Besides @derekbit questions, I am just concerned if any side effects here. Or should we worry about that?

Reply:
@derekbit IIRC it would return `Input/output error`, Sync() would returns `Input/output error` or `read-only file system`
@innobead We might be able to keep `os.RedaDir` as `Readdirnames`, and do `Open/Sync` after `os.ReadDir`. I am trying to test there operations.

Reply:
@mantissahz
I'm fine with the implementation but still confused about why need extra `Sync()` after `Readdirnames`. Making mount point invalid sounds a bit weird.

Reply:
I guess it's for triggering read (`Readdirnames`) and write `Sync` ops. `Readdirnames` may not be able to trigger write op.

Reply:
After discussing @mantissahz, the `os. Readdirnames` still works after detachment, and it is might be due to the cache effect. `Sync()`. Afterwards, using `Sync()` can trigger IO error.
Theoretically, using single `Sync()` without `os. Readdirnames` should be okay. However, according to @mantissahz's test, using single `Sync()` doesn't work. We can investigate it later and add a comment and TODO here.

Reply:
It would not return an error using only `Sync` without `ReadDir` or `Readdirnames` in my testings. It is a weird thing for me.
@derekbit Added comments for using two system calls.
Thanks